stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ApplicationDescriptor.java

```diff
@@ -58,106 +51,36 @@ protected Object onGetChildAt(Application element, int index) {
     return context.getActivitiesList().get(index);
   }
 
-  private ElementContext newElementContext() {
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.ICE_CREAM_SANDWICH) {
-      return new ElementContextICS();
-    } else {
-      LogUtil.w("Running on pre-ICS: must manually reload inspector when Activity changes");
-      return new ElementContextPreICS();
-    }
-  }
-
-  private abstract class ElementContext {
-    public abstract void hook(Application element);
+  private class ElementContext {
+    private final Application mElement;
+    private final ActivityTracker mActivityTracker = ActivityTracker.get();
```

Comment:
nit: we can probably grab the ActivityTracker once in the outer class (ApplicationDescriptor). This is sort of an OCD nit because there will only ever be 1 Application, and thus only 1 ApplicationDescriptor.ElementDescriptor. Still, it's nice to have consistency for when someone else inevitably comes into this class and wonders where they should put more stuff.